lib/Models/Videos.php

```diff
@@ -705,9 +705,12 @@ private static function addEpisodeAcl($episode_id, $add_acl, $acl)
         }
 
         $result = [];
-        foreach ($acl as $entry) {
-            if (in_array($entry['role'], $possible_roles) === false) {
-                $result[] = $entry;
+
+        if (is_array($acl)) {
```

Comment:
I recommend adding this check to `updateAcl' after https://github.com/elan-ev/studip-opencast-plugin/blob/8f576dc16a7af8fdfb316d4c0c8edf36e8671638/lib/Models/Videos.php#L754-L757 to prevent the current acls from being overwritten if the current acls could not be fetched. If the fetch fails the update process should be aborted.

Reply:
Ok, added that check in https://github.com/elan-ev/studip-opencast-plugin/pull/1074/commits/ab14e5de9832260a85f73f903d02bbd04c7fcb80